runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4568,11 +4568,39 @@ done:;
 	ldc2lw(REGISTER_ARGS_LIST)
 	{
 		U_16 index = *(U_16*)(_pc + 1);
-		J9ROMConstantRef *romConstant = (J9ROMConstantRef*)romConstantPoolItem(_literals, index);
-		_pc += 3;
-		_sp -= 2;
-		((U_32*)_sp)[0] = romConstant->slot1;
-		((U_32*)_sp)[1] = romConstant->slot2;
+
+		if (J9CPTYPE_CONSTANT_DYNAMIC == J9_CP_TYPE(J9ROMCLASS_CPSHAPEDESCRIPTION(J9_CLASS_FROM_METHOD(_literals)->romClass), index)) {
+			J9ConstantPool *ramConstantPool = J9_CP_FROM_METHOD(_literals);
+			J9RAMConstantDynamicRef *ramConstant = (J9RAMConstantDynamicRef*)ramConstantPool + index;
+			j9object_t resolvedValue = ramConstant->value;
+
+			if (NULL == resolvedValue) {
+				buildGenericSpecialStackFrame(REGISTER_ARGS, 0);
+				updateVMStruct(REGISTER_ARGS);
+
+				resolvedValue = resolveConstantDynamic(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE);
+		
+				VMStructHasBeenUpdated(REGISTER_ARGS);
+				restoreGenericSpecialStackFrame(REGISTER_ARGS);
+
```

Comment:
Please add the immediate async check - see other resolves for an example

Reply:
async check added